core/models/payload.go

```diff
@@ -295,3 +314,31 @@ func (r *ResponseDetails) ConvertToResponseDetailsView() v1.ResponseDetailsView
 
 	return v1.ResponseDetailsView{Status: r.Status, Body: body, Headers: r.Headers, EncodedBody: needsEncoding}
 }
+
+func (r *ResponseDetails) ConvertToV2ResponseDetailsView() v2.ResponseDetailsView {
+	needsEncoding := false
+
+	// Check headers for gzip
+	contentEncodingValues := r.Headers["Content-Encoding"]
+	if len(contentEncodingValues) > 0 {
+		needsEncoding = true
+	} else {
+		mimeType := http.DetectContentType([]byte(r.Body))
+		needsEncoding = true
+		for _, v := range supportedMimeTypes {
+			if strings.Contains(mimeType, v) {
+				needsEncoding = false
```

Comment:
I agree with you @danielbryantuk. Having gone over it myself, its not as readable as it could be.

I've added comments explaining the functionality of that code to both of the places it occurs. Once we strip out the v1 API code from Hoverfly, I'll refactor this.